routes/api/contacts.js

```diff
@@ -1,25 +1,141 @@
-const express = require('express')
+const express = require("express");
+const { Contacts } = require("../schema");
+const router = express.Router();
+const { validateJoi } = require("./validation");
 
-const router = express.Router()
+router.get("/", async (__, res) => {
+  const contacts = await Contacts.find();
+  try {
+    res.json({
+      status: "success",
+      code: 200,
+      data: contacts,
+    });
+  } catch (error) {
+    res.status(400).json({
+      status: "error",
+      code: 400,
+      message: "Bad Request",
+    });
+  }
+});
 
-router.get('/', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+router.get("/:contactId", async (req, res) => {
+  const { contactId } = req.params;
+  try {
+    const contact = await Contacts.findById(contactId);
+    res.json({
+      status: "success",
+      code: 200,
+      data: contact,
+      message: "Contact has been found",
+    });
+  } catch (error) {
+    res.status(400).json({
+      status: "error",
+      code: 400,
+      message: "Bad Request",
+    });
+  }
+});
 
-router.get('/:contactId', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+router.post("/", async (req, res) => {
+  const body = req.body;
+  const contact = new Contacts(body);
+  const validate = validateJoi(contact);
```

Comment:
```suggestion
  const body = req.body;
  const validate = validateJoi(contact);
  if (validate.error) return res.status(400).send(some error);
  
  const contact = new Contacts(body);
  
```

Reply:
Powinno być tak że jak otrzymasz dane z frontendu to najpierw je walidujesz i sprawdzasz czy jest błąd. Musisz tak mieć zrobioną walidację że gdy coś pójdzie nie tak to ona zwróci informacje o błędzie że na przykład brakuje czegoś albo że coś jest nie tak jak ja powinno być bo na przykład email nie jest e-mailem.